daft/ai/openai/protocols/prompter.py

```diff
@@ -108,22 +118,52 @@ def _process_message_content(self, msg: Any) -> dict[str, str]:
             encoded_content = f"data:{mime_type};base64,{base64_string}"
         else:
             raise ValueError(f"Unsupported content type in prompt: {type(msg)}")
+
         # Determine if it's an image or generic file based on MIME type
         if mime_type.startswith("image/"):
-            return {"type": "input_image", "image_url": encoded_content}
+            if self.use_chat_completions:
+                return {"type": "image_url", "image_url": {"url": encoded_content}}  # type: ignore[dict-item]
+            else:
+                return {"type": "input_image", "image_url": encoded_content}
```

Comment:
This is fine for now, you could also use singlemethoddispatch and register each time with a fallback, this way you're leveraging python's type switching/dispatch rather than using our own isinstance checks.